backend/pikpak/pikpak.go

```diff
@@ -128,6 +129,7 @@ func pikpakAuthorize(ctx context.Context, opt *Options, name string, m configmap
 
 // Register with Fs
 func init() {
+	pphashType = hash.RegisterHash("pikpak", "PikPakHash", 40, sha1.New)
```

Comment:
Hashes don't take a length function so short answer would be that you can't :-(

You could read all the data in your hash function before choosing the block size - that doesn't sound like a good solution to me though.

However I see that there are only 8 possible block sizes from 0x40000 to 0x200000 so you could register 8 hash functions but I don't think this will work well with the rclone core as it won't know which hash function to choose.

We've thought about size dependent hashes before when thinking about multipart Etags for S3. We decided not to use them though.

I don't think this is possible with rclone's hashing framework as it stands today :-(

Looking at the code for pikpak I see we have an MD5 which seems to work, so I'd say we don't need the support for the pikpak hash in the rclone core as MD5 is very good for detecting errors.

What is this hash used for? If it is just a content hash then I think we should probably remove the broken sha1 implementation from pikpak and just use MD5. It may have other uses I'm not aware of though.

What do you think?